project/Dependencies.scala

```diff
@@ -18,7 +18,15 @@
 import sbt._
 
 object Dependencies {
-  lazy val common = core ++ infoset ++ test
+
+  // Note cli added to commmon on purpose.
+  // Due to some obscure bug, updateClassifers was not pulling the
+  // dependencies for the daffodil-cli module into lib_managed properly.
+  // (It found the source jars and javadoc jars, but not the regular code jar for
+  // Scallop aka org.rogach.)
+  // So compilation would fail for daffodil-cli from missing org.rogach library.
+  // Added to here forces these dependencies to be resolved. 
+  lazy val common = core ++ infoset ++ test ++ cli
```

Comment:
So I'm pretty sure this is a bug with SBT. The steps I've found to reliably reproduce this:
```bash
git clean -xdf
sbt daffodil-cli/compile # compiles all sources, including daffodil-cli
sbt clean # cleans everything except daffodil-cli subproject (since we don't aggregate it)
sbt updateClassifiers daffodil-cli/updateClassifiers # update classifiers, note this does not download cli dependencies (sbt bug?)
sbt daffodil-cli/compile # fails due to missing cli dependencies
```
I suspect the issue here is that ``sbt clean`` empties the lib_managed directory. But daffodil-cli thinks the lib_managed directory is up to date since daffodil-cli was never cleaned. Since daffodil-cli thinks lib_managed is correct, it doesn't pull the dependencies in lib_managed, resulting in a failure of missing dependencies. I've confirmed this issue exists with sbt 1.1.1, so an sbt update won't help. This feels like a bug in sbt, but a workaround is to just run ``sbt daffodil-cli/clean`` after running ``sbt clean``. This removes the daffodil-cli state that thinks lib_managed is correct and causes it to pull dependencies. I've confirmed that the above works if I insert that command.

If the ``sbt daffodil-cli/clean`` solution seems reasonable and works for you, I'd prefer to revert this change and not have the CLI be a dependency of everything (for example, there's no reason to pull in CLI stuff when using the JAPI). Or alternatively, just never use sbt clean and instead use ``git clean -xdf`` and just be careful to make sure all untracked files that are needed are committed.

Reply:
Why doesn't sbt clean automatically clean daffodil-cli like it does all the other subprojects? 

The only exceptional behavior we need/want is for test and that's simply because they take too long to run. We don't want to run the cli tests automatically, but for everything else we *do* want it to act across daffodil-cli as well. 

Or can this not be achieved?

I can revert this change certainly, and the workaround is fine, but treating daffodil-cli as an entirely separate project but still having it be a subproject is awkward.  

Does any sbt task act across all the sub modules including daffodil-cli, or is daffodil-cli entirely separate now?

Reply:
Unfortunately, I'm not sure SBT has a good to way to disable aggregation of a specified command for a subproject. You can either have "do not aggregate this entire subproject" or "do not aggregate a specified command for all subprojects", but you can't combine the two.

However, SBT does have a builtin "integration" test concept. Basically, you put integration tests in ``src/it/{scala,java,resources}``, and then all integration tests can be run via ``sbt it`` or something. This is sortof similar to what we had with the scala-cli stuff, except CLI tests don't share scala/java/resources directories like they did before and can now, but I don't think that really matters with our current usage. There isn't any sharing going on. Also, I'm not sure CLI tests are really "integration" tests, but I'm not sure that really matters. I'm also not sure if we can make it work with debug, which was one of the reasons for getting rid of the scala-cli stuff.

I'll create a bug to look into if this wold work as a better alternative, and allow us to reenable aggregation and avoid these types of issues.